### PR TITLE
support truncated output in markdown examples

### PR DESCRIPTION
**Pull request by mikesbrown**
Extend the markdown info string for `zq-output` to include an optional `head:N` directive, where only the first N lines of output of a `zq-command` are verified against its `zq-output`.

To find these we must extend our AST walk function. Rename `ZQExamplePair` to `ZQExampleInfo`; add a new struct member to hold the number of lines to compare. Do the same for `ZQExampleTest`.

Move the responsibility of actually running the `zq-command` to a `ZQExampleTest` `Run()` method. This also handles any truncation.

Update all remaining examples in `zql/docs`.

### CODE REVIEW

**Comment by mikesbrown** [comment]
#329 could come first if there is a desire to make `make test-system` not run these.

File: tests/zq_example.go [hunk 1]
```diff
@@ -102,28 +89,75 @@ import (
...
 
-// ZQExamplePair holds a ZQ example as found in markdown.
-type ZQExamplePair struct {
-	command *ast.FencedCodeBlock
-	output  *ast.FencedCodeBlock
+// ZQExampleInfo holds a ZQ example as found in markdown.
+type ZQExampleInfo struct {
+	command         *ast.FencedCodeBlock
+	output          *ast.FencedCodeBlock
+	outputLineCount uint8
```
Comment (alfred-landrum) [inline tests/zq_example.go]: No need to be as specific as `uint8`, `uint` is acceptable.

File: tests/zq_example.go [hunk 2]
```diff
@@ -102,28 +89,75 @@ import (
...
-// from a ZQExamplePair.
+// from a ZQExampleInfo.
 type ZQExampleTest struct {
-	Name     string
-	Command  []string
-	Expected string
+	Name            string
+	Command         []string
+	Expected        string
+	OutputLineCount uint8
```
Comment (alfred-landrum) [inline tests/zq_example.go]: No need to be as specific as `uint8`, `uint` is acceptable.

**Comment by mikesbrown** [comment]
Rebase. Incidentally some of the examples' outputs have changed.

**Pull request merged**